### PR TITLE
Refactor/view property

### PR DESCRIPTION
**Pull request by vallo87**
#75 

modules-view responsible for keyboard interaction (esc)
new property-view responsible for opening/closing element
relationship-view responsible for keyboard interaction (esc)
staggered-list fixed
comments & minors

### CODE REVIEW

File: src/Template/Element/Form/relations.twig [hunk 1]
```diff
@@ -112,10 +110,15 @@
...
+                            <div class="modal-view" v-show="isVisible"
+                                {# @keyup.esc.prevent.stop="handleKeyboard($event)"> #}
+
+                                v-hotkey="keyEvents">
+
                                 <div class="modal-box">
                                     <div class="modal-header">
-                                        <button class="close" @click.prevent="hideRelationshipModal()">&times;</button>
+                                        <button class="close"
+                                            @click.prevent="hideRelationshipModal()">&times;</button>
```
Comment (didoda) [inline src/Template/Element/Form/relations.twig]: what is `&times;`? Button labels should be i18n IMHO, i.e.: `{{ __(' ... whatever ...') }}`

File: src/Template/Element/_elements.scss [hunk 2]
```diff
@@ -167,27 +167,30 @@ body {
...
+            will-change: opacity, max-height;
 
-        // .staggered-enter {
-        //     opacity: 0;
-        //     transition: opacity .4s ease-in;
-        //     will-change: opacity;
-        // }
+            &.staggered-enter-active {
+                opacity: 0;
+                // max-height: 0;
```
Comment (didoda) [inline src/Template/Element/_elements.scss]: I'd drop this...

File: src/Template/Element/_elements.scss [hunk 3]
```diff
@@ -167,27 +167,30 @@ body {
...
+                transition: none;
+                will-change: none;
+            }
         }
 
         .related-list{
             .added-related-object {
-                &:first-child {
-                    border-top: 1px solid;
+                // opacity: 0.7;
```
Comment (didoda) [inline src/Template/Element/_elements.scss]: as above, I'd drop this

File: src/Template/Pages/Modules/_modules.scss [hunk 4]
```diff
@@ -77,7 +77,35 @@ body.view-module {
...
+            }
+
+            &.slide-enter-active {
+                max-height: 0;
+            }
+
+            &.slide-enter-to {
+                max-height: 100vh;
+            }
+        }
```
Comment (didoda) [inline src/Template/Pages/Modules/_modules.scss]: Could it be regrouped as follows?
```
&.slide-leave-active, &.slide-enter-to {
    max-height: 100vh;
}
&.slide-leave-to, &.slide-enter-active {
    max-height: 0;
}
```

File: webroot/js/app/components/property-view/property-view.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Templates that uses this component (directly or indirectly):
```
Comment (didoda) [inline webroot/js/app/components/property-view/property-view.js]: typo: `Templates that use` ...

File: webroot/js/app/components/property-view/property-view.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ *  Template/Elements/Form/other_properties.twig
+ *  Template/Elements/Form/core_properties.twig
+ *  Template/Elements/Form/meta.twig
+ *
+ * <property-view> component used for ModulesPage -> View
+ *
+ * Component that wraps group of properties in the object View
+ *
+ * @prop {Boolean} tabOpen
+ * @prop {String} label
```
Comment (didoda) [inline webroot/js/app/components/property-view/property-view.js]: Missing description for props doc

File: webroot/js/app/components/relation-view/relation-view.js [hunk 7]
```diff
@@ -4,12 +4,25 @@
  *
  * <relation-view> component used for ModulesPage -> View
  *
+ * @prop {String} relationName
```
Comment (didoda) [inline webroot/js/app/components/relation-view/relation-view.js]: Missing description for prop

File: webroot/js/app/components/staggered-list.js [hunk 8]
```diff
@@ -12,8 +12,8 @@ Vue.component('staggered-list', {
     template: `
         <transition-group appear
             name="${NAME}"
-            v-on:enter=enter
-            >
+            v-on:enter="enter"
+            v-on:after-enter=afterEnter>
```
Comment (didoda) [inline webroot/js/app/components/staggered-list.js]: Just to be coherent with previously used syntax: `v-on:after-enter="afterEnter"`

File: webroot/js/beditaUI.js [hunk 9]
```diff
@@ -782,24 +746,24 @@ if (typeof a.download == 'undefined') {
...
+//         if ($('.tab2').next().is(":visible")) {
+//             $('.tab2').BEtabsclose();
+//         } else {
+//             $('.tab2').BEtabsopen();
+//         }
 
-    }
-})
+//     }
+// })
```
Comment (didoda) [inline webroot/js/beditaUI.js]: Is this block not necessary? Then let's drop it ^_^

File: webroot/js/beditaUI.js [hunk 10]
```diff
@@ -830,41 +794,41 @@ $(document).on('click', '.showmore', function() {
...
+
+//             var idAttr = $(this).next().attr('id');
+//             if (idAttr != '' && typeof idAttr !== 'undefined') {
+//                 openTabs.push('#' + idAttr);
+//             }
+//         });
+
+//         localStorage.setItem(title, openTabs);
+//     });
+// }
```
Comment (didoda) [inline webroot/js/beditaUI.js]: As above: if not necessary, drop it

File: src/Template/Element/Form/meta.twig [hunk 11]
```diff
@@ -1,20 +1,24 @@
...
-            <th>{{ key }}:</th>
-            <td>{% if val is iterable %}{{ val|json_encode() }}{% else %}{{ val }}{% endif %}</td>
-        </tr>
-        {% endfor %}
-    </tbody>
-</table>
+        <table class="bordered">
+            <tbody>
+                <tr>
+                    <th>id:</th>
```
[the remark this answers is not in the document]
Comment (vallo87) [reply src/Template/Element/Form/meta.twig]: era già così! cmq giusta segnalazione, correggo

File: src/Template/Element/Form/relations.twig (hunk 1 above)
Comment (vallo87) [reply src/Template/Element/Form/relations.twig]: è una x

**Comment by vallo87** [comment]
the issue was related to the new vue delimiters we chose ( <: :> )
i provided the fix in the last commit

**Pull request merged**